src/Aspire.Dashboard/Utils/GlobalizationHelpers.cs

```diff
@@ -1,65 +1,159 @@
+        var localizedCultureNames = new string[]
+        {
+            "en", "cs", "de", "es", "fr", "it", "ja", "ko", "pl", "pt-BR", "ru", "tr", "zh-Hans", "zh-Hant", // Standard cultures for compliance.
+        };
+
+        LocalizedCultures = localizedCultureNames.Select(CultureInfo.GetCultureInfo).ToList();
+
+        AllCultures = GetAllCultures();
+
+        ExpandedLocalizedCultures = GetExpandedLocalizedCultures(LocalizedCultures, AllCultures);
+    }
 
-    public static string[] GetSupportedCultures()
+    private static Dictionary<string, List<CultureInfo>> GetExpandedLocalizedCultures(List<CultureInfo> localizedCultures, List<CultureInfo> allCultures)
     {
-        var supportedCultures = CultureInfo.GetCultures(CultureTypes.AllCultures)
-            .Where(culture => LocalizedCultures.Contains(culture.TwoLetterISOLanguageName) || LocalizedCultures.Contains(culture.Name))
-            .Select(culture => culture.Name)
-            .ToList();
-
-        // Non-standard culture but it is the default in many Chinese browsers. Adding zh-CN allows OS culture customization to flow through the dashboard.
-        supportedCultures.Add("zh-CN");
-        return supportedCultures.ToArray();
+        var dict = new Dictionary<string, List<CultureInfo>>(StringComparers.CultureName);
+        foreach (var localizedCulture in localizedCultures)
+        {
+            var selfAndChildren = new List<CultureInfo>();
+            dict[localizedCulture.Name] = selfAndChildren;
+
+            foreach (var culture in allCultures)
+            {
+                var current = culture;
+                var parentCount = 0;
+
+                // The top-level parent of all cultures is invariant culture.
+                while (current != CultureInfo.InvariantCulture)
+                {
+                    if (current.Equals(localizedCulture))
+                    {
+                        selfAndChildren.Add(culture);
+                        break;
+                    }
+                    if (parentCount >= MaxCultureParentDepth)
+                    {
+                        // A recursion limit ensures there is no chance of an infinite loop from a circular parent chain.
+                        break;
+                    }
+                    parentCount++;
+                    current = current.Parent;
+                }
+            }
+        }
+
+        return dict;
     }
 
-    public static bool TryGetCulture(this ISet<CultureInfo> cultureOptions, CultureInfo culture, bool matchParent, [NotNullWhen(true)] out CultureInfo? matchedCulture)
+    private static List<CultureInfo> GetAllCultures()
     {
-       if (cultureOptions.Contains(culture))
-       {
-           matchedCulture = culture;
-           return true;
-       }
-
-       // this doesn't work for zh as we support two different zh cultures
-       if (matchParent && !StringComparers.CultureName.Equals(culture.TwoLetterISOLanguageName, "zh"))
```

Comment:
Why was `zh` excluded here? While there are two `zh` localized cultures, `zh-CN` always resolves to one via parent.